### PR TITLE
feat: better handling of some 404s

### PR DESCRIPTION
**Pull request by rpcross**
- there are some 18k incorrect archive-at headers, redirect these to correct url
- if a message has been removed from archive return 410 with generic removed message

### CODE REVIEW

**Comment by rjsparks** [comment]
How is the Redirect table going to get populated?

**Comment by rpcross** [comment]
The Redirect table will get populated by a custom script using output from the archived-at scan

File: backend/mlarchive/bin/scan_all.py [hunk 1]
```diff
@@ -175,23 +175,29 @@ def archived_at():
...
-                matches.append(hashcode in url)
-        if matches and not any(matches):
-            print('{},{},{}'.format(message.date.strftime('%m-%d-%Y'),url, hashcode))
+        # problem messages will have one mailarchive archived-at header that does not match
+        if len(archives) == 1 and 'mailarchive' in archives[0] and hashcode not in archives[0]:
+            print('{},{},{}'.format(message.date.strftime('%m-%d-%Y'),archives[0], hashcode))
+            if fix:
+                parts = urlparse(archives[0].strip('<>'))
+                if parts.path:
+                    try:
```
Comment (rjsparks) [inline backend/mlarchive/bin/scan_all.py:197]: The pass in the except bothers me a little. Consider looking to see if key (old) already exists. If so, and new matches, then pass. If new does not match, you should probably find a way to raise it as an issue.

**Comment by dmantryingcode** [comment]
Yes please sorry im new im learning as i go to sky, had no right to be on my system. I don’t even know how you gotth re te

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Review by rjsparks — APPROVED** [review]

**Pull request merged**